linkerd/http/prom/src/stream_label.rs

```diff
@@ -36,26 +22,25 @@ pub trait MkStreamLabel {
 }
 
 pub trait StreamLabel: Send + 'static {
-    type DurationLabels: EncodeLabelSet
-        + Clone
-        + Eq
-        + std::fmt::Debug
-        + std::hash::Hash
-        + Send
-        + Sync
-        + 'static;
-    type StatusLabels: EncodeLabelSet
-        + Clone
-        + Eq
-        + std::fmt::Debug
-        + std::hash::Hash
-        + Send
-        + Sync
-        + 'static;
+    type DurationLabels;
+    type StatusLabels;
```

Comment:
I know this is the main point of this PR but I'm missing how it all ties together. It seems like we incur a ton of boilerplate by not having these enforce a `: LabelSet` bound here.

Where _aren't_ these bounds actually required? Is there a way for us to special-case that?